server/src/test/scala/org/http4s/server/middleware/HttpsRedirectSpec.scala

```diff
@@ -8,50 +8,65 @@ package org.http4s
 package server
 package middleware
 
+import cats.implicits._
 import cats.effect._
 import org.http4s._
 import org.http4s.dsl.io._
 import org.http4s.headers._
+import org.http4s.syntax.all._
 import org.http4s.Uri.{Authority, RegName, Scheme}
-import org.http4s.testing.Http4sLegacyMatchersIO
 
-class HttpsRedirectSpec extends Http4sSpec with Http4sLegacyMatchersIO {
+class HttpsRedirectSpec extends Http4sSuite {
   val innerRoutes = HttpRoutes.of[IO] { case GET -> Root =>
     Ok("pong")
   }
 
   val reqHeaders = Headers.of(Header("X-Forwarded-Proto", "http"), Header("Host", "example.com"))
   val req = Request[IO](method = GET, uri = Uri(path = "/"), headers = reqHeaders)
 
-  "HttpsRedirect" should {
-    "redirect to https when 'X-Forwarded-Proto' is http" in {
-      List(
-        HttpsRedirect(innerRoutes).orNotFound,
-        HttpsRedirect.httpRoutes(innerRoutes).orNotFound,
-        HttpsRedirect.httpApp(innerRoutes.orNotFound)
-      ).map { app =>
-        val resp = app(req).unsafeRunSync()
-        val expectedAuthority = Authority(host = RegName("example.com"))
-        val expectedLocation =
-          Location(
-            Uri(path = "/", scheme = Some(Scheme.https), authority = Some(expectedAuthority)))
-        val expectedHeaders = Headers(expectedLocation :: `Content-Type`(MediaType.text.xml) :: Nil)
-        resp.status must_== Status.MovedPermanently
```

Comment:
The original had 2 assertions and I converted them to 2 tests. It could also be done as
`IO(a).assert.. >> IO(b).assert`
WDYT?

Reply:
I think I'm fine with the `>>` -- it's asserting over related things, and there's not insignificant setup.  But I could go either way.